src/main/java/edu/cornell/kfs/rass/batch/ProposalTranslationDefinition.java

```diff
@@ -11,16 +11,29 @@
 import org.kuali.kfs.module.cg.businessobject.Agency;
 import org.kuali.kfs.module.cg.businessobject.Proposal;
 import org.kuali.rice.core.api.datetime.DateTimeService;
+import org.springframework.beans.factory.InitializingBean;
 
+import edu.cornell.kfs.module.cg.CuCGPropertyConstants;
 import edu.cornell.kfs.rass.RassConstants;
 import edu.cornell.kfs.rass.batch.xml.RassXmlAwardEntry;
 import edu.cornell.kfs.rass.util.RassUtil;
 
-public class ProposalTranslationDefinition extends RassObjectTranslationDefinition<RassXmlAwardEntry, Proposal> {
+public class ProposalTranslationDefinition extends RassObjectTranslationDefinition<RassXmlAwardEntry, Proposal>
+            implements InitializingBean {
 
 	protected BusinessObjectService businessObjectService;
 	protected DateTimeService dateTimeService;
 	protected ParameterService parameterService;
+	protected RassPropertyDefinition grantNumberMapping;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
```

Comment:
Why is this needed, isn't the mapping setup in cu-spring-rass.xml ?

Reply:
The Grant Number mapping is included in the "propertyMappings" list, but I didn't want to loop through the mappings list every time that just this individual mapping was needed. (This one mapping is needed for the modified can-edit check.) I added this afterPropertiesSet() method to set up a new "grantNumberMapping" field automatically once the bean is all configured, without requiring the bean configuration to add the Grant Number mapping twice.